### PR TITLE
E1645. Tree display controller refactor

### PR DESCRIPTION
**Pull request by shubham2892**
Code Refactor on tree_display_controller. Issue 1645

### CODE REVIEW

File: app/controllers/tree_display_controller.rb [hunk 1]
```diff
@@ -124,162 +108,212 @@ def list
...
-  # for child nodes
-  def get_children_node_2_ng
-    childNodes = {}
-    childNodes = if params[:reactParams2][:child_nodes].is_a? String
-                   JSON.parse(params[:reactParams2][:child_nodes])
-                 else
-                   params[:reactParams2][:child_nodes]
-                 end
-    tmpRes = {}
+  def available_condition2?(instructor_id, child)
```
Comment (yangsong8) [inline app/controllers/tree_display_controller.rb]: could you figure out what does this method do and give it a better name? Thanks,

**Comment by Winbobob** [comment]
Please do following two things:
 - Sync forked repo and remove the potential conflicts
```
git remote add upstream https://github.com/expertiza/expertiza.git
git remote -v
git fetch upstream
git merge upstream/master
```
 -  Fix the broken test cases caused by your refactor

**Comment by shubham2892** [comment]
@Winbobob : Synced the repo with the upstream master branch. Also, All the test cases of `tree_display_controller_spec.rb` are passing.

**Comment by Winbobob** [comment]
Some method names are confusing, such as `session_last_open_tab` and `set_session_last_open_tab`. It will be better if either changing the method name or add some more comments.

**Comment by saratkavuru** [comment]
@Winbobob I have added some comments to make the methods more obvious. Kindly, review and get back to us again if any  further changes are required .

**Comment by Winbobob** [comment]
LGTM, good job!

**Pull request merged**